database/dataDefinitionQueries.sql

```diff
@@ -84,6 +84,8 @@ CREATE TABLE `UserPreference` (
  `MinAge` int,
  `MaxAge` int,
  `More` varchar(255),
+ `Distance` varchar(100),
```

Comment:
There are already zipCode and distancePreference columns in the user table. Is there a specific reason why these were added here? Otherwise I think it should be moved to the user table

Reply:
Honestly I forgot they were there. I can think of a couple of options here:

1. In the current frontend design, I have the distance and zip code in the filters, so it may make sense for them to be treated as preferences like other filter inputs. Advantage: user can easily search different distances/zip codes and save as preferences. If we're going for this, I can make the change for the user table/features.
2. To stick to the original plan, I can remove the distance and zipcode from the filters and fix them using the values from the user table.
3. Similar to 2, but allows users to change the distance and zip code when searching for pets (but users can't save the values they searched)

Do you have a suggestion for which approach we should take? I'm fine with all of them so let me know!

Reply:
Having distancePreference in the user table made sense at the time of creation, but I agree that it makes more sense in the preferences given the way other implementation turned out. I think that would be good to leave it as you implemented (option 1).

Once concern I have is that the user's address isn't really used at all in that case. Should we remove the address entirely from the User table then? If we did that, we would have to make some changes to the user creation form as well. If we allow the user to enter their address, and they then need to add their zip code again in the filters, they may question why their address was needed at all.

So I think there's two paths you could go down. The first would be to remove the address entirely from the User table. This would include removing the address from account creation and the profile page. The second option would be to let the user enter their address again, and them maybe automatically save the zip code as a user preference. There are probably other good options too that I'm not thinking of.

Again, I agree that your current implementation of distancePreference is best, so I'm only talking about the address. What do you think is the best route forward?

Reply:
I’m glad you brought this up, I could use some advice! The address is actually used in the query given these conditions: the zip code in the filters is the same as the user’s zip code, and we find a valid lat/lng location for the zip code (I assumed the address matches the zip code, this basically allows users to search for other zip codes). I thought this provides a little bit more precision to the calculated distance ( doesn’t matter a lot to the users in my opinion). But I couldn’t find a good way to show this on the pets page. I may prefer to remove the address entirely cause it simplifies the code and as a user I probably wouldn’t want to provide my address if it doesn’t feel necessary. Would appreciate your thoughts on this!

Reply:
Yeah that analysis is pretty spot on imo. Might as well do away with full address for users in favor of just a zip code. The added precision is unnecessary I think. Would you be able to make the appropriate changes in this PR? Probably would include changing the table, removing address references from requests in backend, removing address section in create account form in favor of just a zip. I think that's about it

Reply:
Updated! Ready for review :)